packages/curve-ui-kit/src/hooks/useTokenBalance.tsx

```diff
@@ -1,5 +1,5 @@
 import { type Address, ethAddress, formatUnits } from 'viem'
-import { useBalance } from 'wagmi'
+import { useBalance } from '@ui-kit/features/connect-wallet/lib/wagmi/hooks'
```

Comment:
Unfortunately `useBalance` no longer supports the `token` parameter, and thus only fetches the balance for the native chain currency, like ETH for Ethereum. Check out https://wagmi.sh/react/guides/migrate-from-v1-to-v2#deprecated-usebalance-token-parameter

We should re-implement this hook as directed by the docs, by using `useReadContracts` in combination with `erc20Abi` from `viem`.

I think they did this because not all ERC20 balances follow the spec, and/or decimals may vary